relayer/pkg/starknet/chain_client.go

```diff
@@ -0,0 +1,106 @@
+package starknet
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/NethermindEth/juno/core/felt"
+	starknetrpc "github.com/NethermindEth/starknet.go/rpc"
+	gethrpc "github.com/ethereum/go-ethereum/rpc"
+)
+
+// type alias for readibility
+type FinalizedBlock = starknetrpc.Block
+
+// used to create batch requests
+type StarknetBatchBuilder interface {
+	RequestBlockByHash(h *felt.Felt) (StarknetBatchBuilder, error)
+	// RequestBlockByNumber(id uint64) (StarknetBatchBuilder, error)
```

Comment:
will be uncommented as the corresponding methods in the chain client are implemented